smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -65,6 +70,140 @@ void writeMembers(TypeScriptWriter writer, Shape shape) {
+            throw new CodegenException(String.format(
+                "CollectionFilterSensitiveLog attempted for %s while it was not required",
+                memberShape.getType()
+            ));
+            // For quick-fix in case of high severity issue:
+            // comment out the exception above and uncomment the line below.
+            // writer.write("item => item");
+        }
+    }
+
+    /**
+     * Recursively writes filterSensitiveLog for MapShape.
+     */
+    void writeMapFilterSensitiveLog(TypeScriptWriter writer, MemberShape mapMember) {
+        // Reducer is common to all shapes.
+        writer.openBlock("(acc: any, [key, value]: [string, $T]) => ({", "}), {}",
+            symbolProvider.toSymbol(mapMember),
+            () -> {
+                writer.write("...acc,");
+                Shape memberShape = model.expectShape(mapMember.getTarget());
+                if (memberShape instanceof StructureShape) {
+                    // Call filterSensitiveLog on Structure.
+                    writer.write("[key]: $T.filterSensitiveLog(value),",
+                        symbolProvider.toSymbol(mapMember));
+                } else if (memberShape instanceof CollectionShape) {
+                    writer.openBlock("[key]: value.map(", "),",
+                        () -> {
+                            MemberShape collectionMember = ((CollectionShape) memberShape).getMember();
+                            writeCollectionFilterSensitiveLog(writer, collectionMember);
+                        }
+                    );
+                } else if (memberShape instanceof MapShape) {
+                    writer.openBlock("[key]: Object.entries(value).reduce(", "),",
+                        () -> {
+                            MemberShape nestedMapMember = ((MapShape) memberShape).getValue();
+                            writeMapFilterSensitiveLog(writer, nestedMapMember);
+                        }
+                    );
+                } else {
+                    // This path should not reach because of recursive isIterationRequired.
+                    throw new CodegenException(String.format(
+                        "MapFilterSensitiveLog attempted for %s while it was not required",
+                        memberShape.getType()
+                    ));
+                    // For quick-fix in case of high severity issue:
+                    // comment out the exception above and uncomment the line below.
+                    // writer.write("[key]: value,");
+                }
+            }
+        );
+    }
+
+    void writeFilterSensitiveLog(TypeScriptWriter writer, Shape shape) {
+        writer.write("...obj,");
+        for (MemberShape member : members) {
+            Shape memberShape = model.expectShape(member.getTarget());
+            String memberName = TypeScriptUtils.sanitizePropertyName(symbolProvider.toMemberName(member));
```

Comment:
It looks like the right side of this statement is shared with code in `writeMembers`, can it be made its own function so they don't get out of sync?

Reply:
Fixed in commit https://github.com/awslabs/smithy-typescript/pull/170/commits/3c73a41051d51594e0064e20281e7be634f7bcb3